examples/flutter_gallery/lib/demo/material/bottom_app_bar_demo.dart

```diff
@@ -13,291 +13,311 @@ class BottomAppBarDemo extends StatefulWidget {
-        new Text(
-          'Floating action button',
-          style: Theme.of(context).textTheme.title,
-        ),
-        buildFabShapePicker(),
-        buildFabLocationPicker(),
-        const Divider(),
-        new Text(
-          'Bottom app bar options',
-          style: Theme.of(context).textTheme.title,
-        ),
-        buildBabColorPicker(),
-        new CheckboxListTile(
-          title: const Text('Enable notch'),
-          value: notchEnabled,
-          onChanged: (bool value) {
-            setState(() {
-              notchEnabled = value;
-            });
-          },
-          controlAffinity: ListTileControlAffinity.leading,
-        ),
-      ],
-    );
-  }
+class _ChoiceValue<T> {
+  const _ChoiceValue({ this.value, this.title, this.label });
 
-  Widget buildFabShapePicker() {
-    return new Row(
-      mainAxisAlignment: MainAxisAlignment.spaceBetween,
-      children: <Widget>[
-        const SizedBox(width: 96.0,
-          child: const Text('Shape: '),
-        ),
-        new Expanded(
-          child: new Padding(
-            padding: const EdgeInsets.all(8.0), 
-            child: new RaisedButton(
-              child: const Text('Change shape'),
-              onPressed: () {
-                setState(() {
-                  fabShapeIndex = (fabShapeIndex + 1) % _fabShapeConfigurations.length;
-                });
-              },
-            ),
-          ),
-        ),
-      ],
-    );
-  }
+  final T value;
+  final String title;
+  final String label; // For the Semantics widget that contains title
 
-  Widget buildFabLocationPicker() {
-    return new Row(
-      mainAxisAlignment: MainAxisAlignment.spaceBetween,
-      children: <Widget>[
-        const SizedBox(
-          width: 96.0,
-          child: const Text('Location: '),
-        ),
-        new Expanded(
-          child: new Padding(
-            padding: const EdgeInsets.all(8.0), 
-            child: new RaisedButton(
-              child: const Text('Move'),
-              onPressed: () {
-                setState(() {
-                  fabLocationIndex = (fabLocationIndex + 1) % _fabLocationConfigurations.length;
-                });
-              },
-            ),
-          ),
-        ),
-      ],
-    );
-  }
+  @override
+  String toString() => '$runtimeType("$title")';
+}
 
-  Widget buildBabColorPicker() {
-    final List<Widget> colors = <Widget> [
-      const Text('Color:'),
-    ];
-    for (Color color in babColors) {
-      colors.add(
-        new Semantics(
-          label: 'Set Bottom App Bar color to ${colorToName[color]}',
-          container: true,
-          child: new Row(children: <Widget> [
-            new Radio<Color>(
-              value: color,
-              groupValue: babColor,
-              onChanged: (Color color) {
-                setState(() {
-                  babColor = color;
-                });
-              },
+class _RadioItem<T> extends StatelessWidget {
```

Comment:
This pattern (generic radio item and generic choice value) seems useful enough that I'm wondering how it could be part of the framework (or at least the docs) near Radio.